spark/src/main/scala/io/delta/tables/execution/DeltaTableOperations.scala

```diff
@@ -112,6 +114,48 @@ trait DeltaTableOperations extends AnalysisHelper { self: DeltaTable =>
     toDataset(sparkSession, restore)
   }
 
+  protected def executeClone(
+      table: DeltaTableV2,
+      target: String,
+      replace: Boolean,
+      properties: Map[String, String],
+      versionAsOf: Option[Long] = None,
+      timestampAsOf: Option[String] = None): DeltaTable = withActiveSession(sparkSession) {
+    val sourceIdentifier = table.getTableIdentifierIfExists.map(id =>
+      Identifier.of(id.database.toArray, id.table))
+    val sourceRelation = DataSourceV2Relation.create(table, None, sourceIdentifier)
+
+    val pathBased = new Path(target).isAbsolute()
+    val targetIdentifier = if (pathBased) s"delta.`$target`" else target
+    val targetRelation = UnresolvedRelation(
+      sparkSession.sessionState.sqlParser.parseTableIdentifier(targetIdentifier))
+
+    val maybeTimeTravelSource = if (timestampAsOf.isDefined || versionAsOf.isDefined) {
```

Comment:
```suggestion
    val maybeTimeTravelSource = if (versionAsOf.isDefined || timestampAsOf.isDefined) {
```

Since version is always before timestamp